src/validators/gre-protocol

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2024 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+from sys import argv,exit
+
+if __name__ == '__main__':
+    if len(argv) != 2:
+        exit(1)
+
+    input = argv[1]
+    try:
+        # It's common practice to pass hex for ethtype (eg, 0x8100), so we allow 
+        # other radix prefixes with the int conversion:
+        if int(input, 0) in range(0, 65535):
+            exit(0)
+    except ValueError:
+        pass
+
+    pattern = "!?\\b(ip|ip6|arp|vlan|8021q|8021ad)\\b"
```

Comment:
This script is unnecessary, since the use case is already covered by the `<regex>` validation mechanism, which is a part of the `validate-value` executable and doesn't require spawning a subprocess (and spawning a Python interpreter is a notoriously slow operation).

Reply:
Main reason I did this was for the multi-radix validation (0xNNNN syntax is pretty common for protocols and ethertypes).

The OCaml numeric validator doesn't like radix prefixes, I'll pack this all into a single regex for now - is that something that might be useful for the validator?

Reply:
Also, Is there a neat or "proper" way to offer hex integers in valueHelp, or is it something that is best not done at all?

Right now I'm using:
```
        <valueHelp>
          <format>u32:0x0-0xFFFF</format>
          <description>Ethernet protocol number (hex)</description>
        </valueHelp>
```

This formats correctly in the completion under vbash. I picked through the vyatta-cfg node parser + completion scripts and couldn't immediately spot anywhere that definitely expects a base-10 integer, but I'm not familiar with the API and other internals that may treat those tags differently.

Reply:
>The OCaml numeric validator doesn't like radix prefixes

The only reason for that is that I never thought it would be useful anywhere. It would be quite simple to add, although the UI for it is an interesting question. I made a task about it (https://vyos.dev/T6622), if you have any ideas, please leave a comment there.

Reply:
> This formats correctly in the completion under vbash

This is defined in `/etc/bash_completion.d/vyatta-cfg`

```
  case "$vtype" in
    _*)
      echo -n "${vtype#_}"
      ;;
    txt)
      echo -n '<text>'
      ;;
    u32)
      echo -n '<0-4294967295>'
      ;;
    u32:*)
      echo -n "<${vtype##u32:}>"
      ;;
   ...
```


Using 

```
    <valueHelp>
      <format>u32:0x0-0xffff</format>
      <description>Numeric IP port</description>
    </valueHelp>
```

will yield `<0x0-0xffff>         Numeric IP port`

and 

```
    <valueHelp>
      <format>_0x0-0xffff</format>
      <description>Numeric IP port</description>
    </valueHelp>
```

will yield `0x0-0xffff         Numeric IP port`